src/App.tsx

```diff
@@ -15,71 +16,82 @@ export const goods = [
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  clearGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {selectedGood
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.clearGood}
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                key={good + goods.indexOf(good)}
+                className={
+                  classNames({
+                    'has-background-success-light': good === selectedGood,
+                  })
+                }
+              >
+                <td>
+                  <button
+                    data-cy={good === selectedGood
+                      ? 'RemoveButton'
+                      : 'AddButton'}
+                    type="button"
+                    onClick={good === selectedGood
+                      ? this.clearGood
+                      : () => this.setState({ selectedGood: good })}
+                    className={
+                      classNames({
+                        button: true,
+                        'is-info': good === selectedGood,
+                      })
+                    }
+                  >
+                    {good === selectedGood
+                      ? '-'
+                      : '+'}
+                  </button>
```

Comment:
to avoid so many ternary operators, u can create separate button for each cases.